packages/mdx/test/compile.js

```diff
@@ -858,11 +858,13 @@ test('jsx', async () => {
     [
       '/*@jsxRuntime automatic @jsxImportSource react*/',
       'function _createMdxContent(props) {',
-      '  const _components = props.components || ({}), _component0 = "a-b";',
+      '  const _components = Object.assign({',
+      '    "a-b": "a-b"',
+      '  }, props.components), _component0 = _components["a-b"];',
       '  return <>{<_component0></_component0>}</>;',
       '}',
       'function MDXContent(props = {}) {',
-      '  const _component0 = "a-b", {wrapper: MDXLayout} = props.components || ({});',
+      '  const _component0 = _components["a-b"], {wrapper: MDXLayout} = props.components || ({});',
```

Comment:
Something seems to go wrong here in `MDXContent`, which only needs the wrapper. `_component0` does not need to be retrieved? I don’t believe anything other than the wrapper needs to be retrieved.

It also looks like this would throw: `_components` isn’t defined yet! Perhaps it is a good addition to test “invalid identifiers” can run too (here: https://github.com/mdx-js/mdx/blob/69a15b76f15136a33f2f9fe947a605e97d28fc6c/packages/mdx/test/evaluate.js#L328)?

Reply:
Looks like it indeed throws, here: https://github.com/mdx-js/mdx/runs/7842620509?check_suite_focus=true#step:5:31.

Reply:
@wooorm Yep, I noticed that! Trying to find the code path that's injecting this variable inside layouts. Really want it to _not_ to that 😆

Reply:
Update: figured it out! No longer injects `const _component0` in layouts 👍